diff --git cc/dual_net/tf_dual_net.cc cc/dual_net/tf_dual_net.cc
index 3661b9b..c885304 100644
--- cc/dual_net/tf_dual_net.cc
+++ cc/dual_net/tf_dual_net.cc
@@ -51,6 +51,9 @@ class TfDualNet : public DualNet {
     explicit TfWorker(const tensorflow::GraphDef& graph_def)
         : batch_capacity_(0) {
       tensorflow::SessionOptions options;
+      options.config.set_intra_op_parallelism_threads(1);
+      options.config.set_inter_op_parallelism_threads(0);
+      options.config.set_use_per_session_threads(false);
       options.config.mutable_gpu_options()->set_allow_growth(true);
       session_.reset(tensorflow::NewSession(options));
       TF_CHECK_OK(session_->Create(graph_def));
